Modix.Data/Models/Core/MessageEntity.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace Modix.Data.Models.Core
+{
+    public class MessageEntity
+    {
+        [Key]
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public long Id { get; set; }
```

Comment:
No surrogate keys, please. `MessageId` should be the primary key, as it uniquely identifies every single message, anywhere.

Have a look at the other tracking tables for reference, if you want.